cliquet/tests/test_initialization.py

```diff
@@ -60,13 +60,30 @@ def test_set_the_project_name_from_settings_even_if_specified(self):
         self.assertEqual(config.registry.settings['cliquet.project_name'],
                          'kinto')
 
-    def test_overriden_default_settings(self):
+    def test_overriden_new_default_settings(self):
         config = Configurator()
         defaults = {'cliquet.paginate_by': 102}
         cliquet.initialize(config, '0.0.1', 'project_name',
                            default_settings=defaults)
         self.assertEqual(config.registry.settings['cliquet.paginate_by'], 102)
 
+    def test_overriden_empty_default_settings(self):
+        config = Configurator()
+        defaults = {'cliquet.project_name':  'cliquet'}
+        cliquet.initialize(config, '0.0.1', 'project_name',
+                           default_settings=defaults)
+        self.assertEqual(config.registry.settings['cliquet.project_name'],
+                         'cliquet')
```

Comment:
But ``'cliquet.paginate_by'`` is set to None and ``'cliquet.project_name'`` to ``''`` in DEFAULT_SETTINGS, so both tests cover the current ``if not settings.get(key)`` in the code